bdk_core/src/chain_graph.rs

```diff
@@ -32,25 +32,38 @@ impl<I> Default for ChainGraph<I> {
     }
 }
 
+impl<I> ChainGraph<I> {
+    pub fn chain(&self) -> &sparse_chain::SparseChain<I> {
+        &self.chain
+    }
+
+    pub fn graph(&self) -> &TxGraph {
+        &self.graph
+    }
+}
+
 impl<I: ChainIndex> ChainGraph<I> {
     pub fn insert_tx(
         &mut self,
         tx: Transaction,
-        index: I,
-    ) -> Result<bool, sparse_chain::InsertTxErr> {
-        let changed = self.chain.insert_tx(tx.txid(), index)?;
-        self.graph.insert_tx(tx);
-        Ok(changed)
+        index: Option<I>,
```

Comment:
Hmm interesting choice making this an `Option`. I think it's a fine choice. Is there a situation you had in mind that you would want to add a transaction without knowing its height.

Reply:
Yes, I think it will be cool to explicitly add evicted transactions. I.e. helps for listing transactions that a certain one has replaced.